# a script to generate the BFAR report - to generate the report, in the command line type: rmarkdown::render("scripts/BFAR_generator.R")
source("scripts/04_divesummary.R")
library(dplyr)

#' ---
#' title: Annual Report for Gratuitous Permit No. FBP-0079-14
#' author: Michelle Stuart
#' date: 2018-03-13
#' output:
#' word_document:
#' fig_width: 5
#' fig_height: 5
#' fig_caption: true
#' ---

#' ### Introduction
#' Under the project “Effects of low population density on reef fish
#' connectivity,” we conducted fieldwork and collected anemonefish samples in
#' Leyte from May 1 to June 16, 2016. The primary objective was to collect
#' tissue samples (fin clips) from individuals of Amphiprion clarkii at each of
#' 15 locations. In this season, we sampled 15 sites in Albuera and Bay Bay
#' City, Leyte.
#' We are now beginning laboratory analysis of the collected specimens. We are
#' using genotyping-by-sequencing methods to genotype each specimen at a number
#' of Single Nucleotide Polymorphisms (SNPs). This will allow us to match
#' parents and offspring and to identify when we recapture the same fish in a
#' different field season. This information will allow us to determine whether
#' small populations are self-persistent or whether they rely on surrounding
#' populations for persistence (network persistence).

#'### Inventory
#' Note: All samples are tissue clips from the caudal fin.

# generate a table for the bfar report
# in the field, load 2017 dive data from saved database
dive <- read.csv("data/diveinfo.csv", stringsAsFactors = F) %>% 
  filter(grepl("2017", date)) %>% 
  select(dive_table_id, site, municipality)
# in the field, load the anemones that correspond to those dives
anem <- read.csv("data/anemones.csv", stringsAsFactors = F) %>% 
  filter(dive_table_id %in% dive$dive_table_id) %>% 
  select(anem_table_id, dive_table_id) %>% 
  mutate(dive_table_id = as.integer(dive_table_id), 
    anem_table_id = as.integer(anem_table_id))
anem <- left_join(anem, dive, by = "dive_table_id")
rm(dive)
# in the field, load the fish that correspond to those anemones
fish <- read.csv("data/clownfish.csv", stringsAsFactors = F) %>% 
  filter(anem_table_id %in% anem$anem_table_id, 
    !is.na(fin_id), fin_id != "NULL") %>% 
  select(fish_table_id, anem_table_id, fin_id)
fish <- left_join(fish, anem, by = "anem_table_id")
rm(anem)
bfar <- fish %>% 
  group_by(site, municipality) %>%
  summarise(samples = n())
bfar <- bfar %>% 
  mutate(Province = "Leyte") %>% 
  rename(Municipality = municipality, 
    Site = site, 
    Samples = samples) %>% 
  select(Province, Municipality, Site, Samples)
  


#' 
#' 
#' #' ### Dynamic comments
#' #' We can get fancy and actually dynamically generate some commentary around these results.  That is we can auto-fill parts of our document text
#' #' with objects from the R environment.  This could be useful with analyses that involve stochastic 
#' #' elements changing from run to run like random forest.  Or any analysis where results are subject to change.    
#' #' <br>  
#' #' `r rf$confusion[1,1]` cars are correctly classified as 0.  
#' #' `r rf$confusion[2,2]` cars are correctly classified as 1.  
#' #' `r rf$confusion[1,2]` cars are misclassified as 1.  
#' #' `r rf$confusion[2,1]` cars are misclassified as 0.  
#' #' 
#' #' These numbers were generated by wrapping the R expression to excute into the ticks like so: 
#' #' I don't know how to write this within a `#'` comment without evaluating it, so I'm documenting here as a
#' #' character string:  
#' #+ echo=F, eval=T
#' cat("#' `r rf$confusion[2,1]` cars are misclassified as 0. ")
#' #' <br>
#' #' 
#' 
